apps/schema/src/components/ProfileManager.vue

```diff
@@ -0,0 +1,343 @@
+
+fetch("../api/profiles")
+  .then((response) => response.json())
+  .then((data) => {
+    profileJson.value = data;
+    profileJson.value.tables.forEach(
+      (t) => (t.columns = t.columns?.filter((c) => !c.inherited))
+    );
+  })
+  .catch((err) => (error.value = err));
+
+const tables = computed(() => {
+  if (profileJson.value) {
+    return profileJson.value.tables?.filter(
+      (t) =>
+        ((!search.value || JSON.stringify(t).includes(search.value)) &&
+          selectedProfiles.value.length == 0) ||
+        selectedProfiles.value.some(
+          (p) =>
+            t.profiles?.includes(p) ||
+            t.columns?.some((c) => c.profiles?.includes(p))
+        )
+    );
+  }
+});
+
+const selectedTables = computed(() => {
+  if (profileJson.value) {
+    return profileJson.value.tables?.filter(
+      (t) => t.selected || t.columns?.some((c) => c.selected)
+    );
+  }
+});
+
+const profiles = computed(() => {
+  const result = [];
+  if (!profileJson.value.tables) {
+    return [];
+  }
+  profileJson.value.tables.forEach((t) => {
+    if (t.profiles) {
+      result.push(...t.profiles);
+    }
+    if (t.columns) {
+      t.columns.forEach((c) => {
+        if (c.profiles) {
+          result.push(...c.profiles);
+        }
+      });
+    }
+  });
+  return Array.from(new Set(result));
+});
+
+const isVisible = function (column) {
+  return (
+    (selectedProfiles?.value.length == 0 ||
+      column.profiles?.some((p) => selectedProfiles?.value.includes(p))) &&
+    (!search.value || JSON.stringify(column).includes(search.value))
+  );
+};
+
+const v = function (value) {
+  if (value == null) {
+    return "";
+  } else if (Array.isArray(value)) {
+    if (value.length > 1) {
+      return `"${value.join(",")}"`;
+    } else {
+      return v(value[0]);
+    }
+  } else {
+    return value;
+  }
+};
+
+const selectAll = function (tableName) {
+  tables.value.forEach((t) => {
+    if (t.name === tableName) {
+      t.columns?.forEach((c) => {
+        c.selected = true;
+      });
+    }
+  });
+};
+
+const deselectAll = function (tableName) {
+  tables.value.forEach((t) => {
+    if (t.name === tableName) {
+      t.columns?.forEach((c) => {
+        c.selected = false;
+      });
+    }
+  });
+};
+
+const convertToCsv = function () {
+  var csv =
+    "tableName,tableExtends,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,validation,semantics,description,profiles";
+  selectedTables.value.forEach((t) => {
+    csv += `${v(t.name)},${v(t.inheritName)},,,,,,,,,${v(t.semantics)},${v(
+      t.descrpition
+    )},${v(t.profiles)}\n`;
+    t.columns
+      ?.filter((c) => c.selected)
+      .forEach((c) => {
+        csv += `${v(t.name)},,${v(c.name)},${v(c.columnType)},${v(c.key)},${v(
+          c.required
+        )},${v(c.refSchema)},${v(c.refTableName)},${v(c.refLink)},${v(
+          c.validation
+        )},${v(c.semantics)},${v(c.description)},${v(c.profiles)}\n`;
```

Comment:
split of  and add test

Reply:
agreed, but this is a poc meant for feature previewing. We will do this when we start maturing the feature.